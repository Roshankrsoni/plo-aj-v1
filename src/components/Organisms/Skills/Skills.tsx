import React from "react";
import BarGraph from "../../Atoms/BarGraph/BarGraph";
import "./Skills.css";
import { useGlobalContext } from "../../../context/GlobalContext";

const Skills = () => {
  const { profileData } = useGlobalContext();
  const { skills } = profileData;
  const { development, design } = skills;
  return (
    <div className="col-md-7 page-1 text-left section skills">
      <div className="section-title h2 bold">Skills</div>

      <div className="subsection">
        <div className="subsection-title uppercase body-text grey3 letterspacing-1">
          Development
        </div>
        <div className="subsection-data">
          {development.map((skill, key) => {
            return (
              <div
                key={key}
                className="skill-bar-wrapper row no-gutters align-items-center"
              >
                <div className="skill-label grey-1 body-text bold col-md-3">
                  {skill.skillName}
                </div>
                <BarGraph
                  class="col-md-9"
                  value={skill.skillValue}
                  currentStep={1}
                  sectionStep={1}
                />
              </div>
            );
          })}
        </div>
      </div>
      <div className="subsection">
        <div className="subsection-title uppercase body-text grey3 letterspacing-1">
          Design
        </div>
        <div className="subsection-data">
          {design.map((skill, key) => {
            return (
              <div
                key={key}
                className="skill-bar-wrapper row no-gutters align-items-center"
              >
                <div className="skill-label grey-1 body-text bold col-md-3">
                  {skill.skillName}
                </div>
                <BarGraph
                  class="col-md-9"
                  value={skill.skillValue}
                  currentStep={1}
                  sectionStep={1}
                />
              </div>
            );
          })}
        </div>
      </div>
    </div>
  );
};

export default Skills;
