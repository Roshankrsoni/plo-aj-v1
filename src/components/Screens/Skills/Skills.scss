.skills-content {
  .subsection-title {
    margin-bottom: 2rem;
  }
  .skill-bar-wrapper {
    display: flex;

    .skill-label {
      width: 20%;
    }
    .bar-graph {
      width: 80%;
    }
    & + .skill-bar-wrapper {
      margin-top: 2rem;
    }
  }
}
