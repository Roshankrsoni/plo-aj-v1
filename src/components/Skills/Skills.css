.skill-bar-wrapper + .skill-bar-wrapper{
margin-top:2rem;
}
.subsection-data{
    margin-top:2rem;
}

.subsection + .subsection{
    margin-top:8rem;
}