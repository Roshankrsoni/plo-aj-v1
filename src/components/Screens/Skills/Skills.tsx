import React from "react";
import BarGraph from "../../Atoms/BarGraph/BarGraph";
import "./Skills.scss";
import { useGlobalContext } from "../../../context/GlobalContext";
import SectionVisibilityHOC from "../../Organisms/SectionVisibilityHOC/SectionVisibilityHOC";
import T from "../../../translations/en_IN";
import constants from "../../../helpers/constants";

const Skills = () => {
  const { profileData } = useGlobalContext();
  const { skills } = profileData;
  const { development, design } = skills;
  return (
    <>
      <div className="section-title h2 bold">{T.SKILLS}</div>
      <div className="subsection">
        <div className="subsection-title uppercase body-text grey5 letterspacing-1">
          {T.DEVELOPMENT}
        </div>
        <div className="subsection-data">
          {development.map((skill, key) => {
            return (
              <div
                key={key}
                className="skill-bar-wrapper row no-gutters align-items-center"
              >
                <div className="skill-label grey-1 body-text bold col-md-3">
                  {skill.skillName}
                </div>
                <BarGraph
                  className="col-md-9"
                  value={skill.skillValue}
                  animate={true}
                />
              </div>
            );
          })}
        </div>
      </div>
      <div className="subsection">
        <div className="subsection-title uppercase body-text grey5 letterspacing-1">
          {T.DESIGN}
        </div>
        <div className="subsection-data">
          {design.map((skill, key) => {
            return (
              <div
                key={key}
                className="skill-bar-wrapper row no-gutters align-items-center"
              >
                <div className="skill-label grey-1 body-text bold col-md-3">
                  {skill.skillName}
                </div>
                <BarGraph
                  className="col-md-9"
                  value={skill.skillValue}
                  animate={true}
                />
              </div>
            );
          })}
        </div>
      </div>
    </>
  );
};

export default SectionVisibilityHOC(Skills, constants.classNames.SKILLS);
