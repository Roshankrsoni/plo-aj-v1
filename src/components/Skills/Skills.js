import { useEffect, useState } from 'react';
import BarGraph from '../BarGraph/BarGraph'
import './Skills.css'
// import leftPaneData from '../LeftPane/leftPaneData'

function Skills(props) {

    let [sectionStep, setSectionStep] = useState();
    useEffect(() => {
        let key = 0;
        props.leftPaneData.forEach(ele => {
            if (ele.label === 'Skills') {
                key = ele.key;
            }
        })
        setSectionStep(parseInt(key));
    }, [props.leftPaneData])
    return <div className={"show-on-scroll col-md-7 page-1 text-left section skills " + props.class} >
        <div className="section-title grey4 h2 bold">Skills</div>

        {
            props.dbData.skills ? <div className="subsection">
                <div className="subsection-title uppercase body-text grey3 letterspacing-1">Development</div>
                <div className="subsection-data">
                    {
                        props.dbData.skills.development.map((ele, key) => {
                            return <div key={key} className="skill-bar-wrapper row no-gutters align-items-center">
                                <div className="skill-label grey-1 body-text bold col-md-3">{ele.skillName}</div>
                                <BarGraph class="col-md-9" value={ele.skillValue} currentStep={props.currentStep} sectionStep={sectionStep} />
                            </div>
                        })
                    }

                </div>
            </div>
                :
                ''
        }
        {
            props.dbData.skills ? <div className="subsection">
                <div className="subsection-title uppercase body-text grey3 letterspacing-1">Design</div>
                <div className="subsection-data">
                    {
                        props.dbData.skills.design.map((ele, key) => {
                            return <div key={key} className="skill-bar-wrapper row no-gutters align-items-center">
                                <div className="skill-label grey-1 body-text bold col-md-3">{ele.skillName}</div>
                                <BarGraph class="col-md-9" value={ele.skillValue} currentStep={props.currentStep} sectionStep={sectionStep} />
                            </div>
                        })
                    }

                </div>
            </div>
                :
                ''

        }
    </div>

}

export default Skills