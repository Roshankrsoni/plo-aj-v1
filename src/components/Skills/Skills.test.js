import React from 'react';
import { shallow } from 'enzyme';
import Skills from './Skills';

import dummyData from '../TestData/testData.json'

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Skills dbData={dummyData} />);
        expect(component).toMatchSnapshot();
    });
});