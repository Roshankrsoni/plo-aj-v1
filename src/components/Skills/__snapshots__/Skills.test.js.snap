// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="show-on-scroll col-md-7 page-1 text-left section skills undefined"
>
  <div
    className="section-title grey4 h2 bold"
  >
    Skills
  </div>
  <div
    className="subsection"
  >
    <div
      className="subsection-title uppercase body-text grey3 letterspacing-1"
    >
      Development
    </div>
    <div
      className="subsection-data"
    >
      <div
        className="skill-bar-wrapper row no-gutters align-items-center"
        key="0"
      >
        <div
          className="skill-label grey-1 body-text bold col-md-3"
        >
          HTML
        </div>
        <BarGraph
          class="col-md-9"
          value="80"
        />
      </div>
      <div
        className="skill-bar-wrapper row no-gutters align-items-center"
        key="1"
      >
        <div
          className="skill-label grey-1 body-text bold col-md-3"
        >
          CSS
        </div>
        <BarGraph
          class="col-md-9"
          value="80"
        />
      </div>
      <div
        className="skill-bar-wrapper row no-gutters align-items-center"
        key="2"
      >
        <div
          className="skill-label grey-1 body-text bold col-md-3"
        >
          JavaScript
        </div>
        <BarGraph
          class="col-md-9"
          value="80"
        />
      </div>
      <div
        className="skill-bar-wrapper row no-gutters align-items-center"
        key="3"
      >
        <div
          className="skill-label grey-1 body-text bold col-md-3"
        >
          React Js
        </div>
        <BarGraph
          class="col-md-9"
          value="80"
        />
      </div>
      <div
        className="skill-bar-wrapper row no-gutters align-items-center"
        key="4"
      >
        <div
          className="skill-label grey-1 body-text bold col-md-3"
        >
          Salesforce LWC
        </div>
        <BarGraph
          class="col-md-9"
          value="60"
        />
      </div>
    </div>
  </div>
  <div
    className="subsection"
  >
    <div
      className="subsection-title uppercase body-text grey3 letterspacing-1"
    >
      Design
    </div>
    <div
      className="subsection-data"
    >
      <div
        className="skill-bar-wrapper row no-gutters align-items-center"
        key="0"
      >
        <div
          className="skill-label grey-1 body-text bold col-md-3"
        >
          Adobe Illustrator
        </div>
        <BarGraph
          class="col-md-9"
          value="40"
        />
      </div>
      <div
        className="skill-bar-wrapper row no-gutters align-items-center"
        key="1"
      >
        <div
          className="skill-label grey-1 body-text bold col-md-3"
        >
          Figma
        </div>
        <BarGraph
          class="col-md-9"
          value="80"
        />
      </div>
      <div
        className="skill-bar-wrapper row no-gutters align-items-center"
        key="2"
      >
        <div
          className="skill-label grey-1 body-text bold col-md-3"
        >
          Adobe XD
        </div>
        <BarGraph
          class="col-md-9"
          value="50"
        />
      </div>
    </div>
  </div>
</div>
`;
